FindNetCDF.cmake: use user provided paths also if MPI is not found

In `FindNetCDF.cmake`, we allow the user to suggest search paths via `NetCDF_C_PATHS` and `NetCDF_Fortran_PATHS` if the corresponding MPI was found. However, if that's not the case, we don't use the user provided paths.

Suggestion: change lines 46-54 of FindNetCDF.cmake from
```
        if (MPI_${NCDFcomp}_FOUND)
            initialize_paths (NetCDF_${NCDFcomp}_PATHS
                              INCLUDE_DIRECTORIES ${MPI_${NCDFcomp}_INCLUDE_PATH}
                              LIBRARIES ${MPI_${NCDFcomp}_LIBRARIES})
          find_package_component(NetCDF COMPONENT ${NCDFcomp}
                                 PATHS ${NetCDF_${NCDFcomp}_PATHS})
        else()
          find_package_component(NetCDF COMPONENT ${NCDFcomp})
        endif ()
```
to
```
        if (MPI_${NCDFcomp}_FOUND)
            initialize_paths (NetCDF_${NCDFcomp}_PATHS
                              INCLUDE_DIRECTORIES ${MPI_${NCDFcomp}_INCLUDE_PATH}
                              LIBRARIES ${MPI_${NCDFcomp}_LIBRARIES})
        endif ()
        find_package_component(NetCDF COMPONENT ${NCDFcomp}
                               PATHS ${NetCDF_${NCDFcomp}_PATHS})
```

I can submit a ninja-PR if this seems acceptable.


Note: the issue is not within scorpio, per se. When my cmake parses scorpio, it finds MPI components, and everything works. However, due to the search paths of cmake, the same FindNetCDF.cmake file is used to find NetCDF for the cprnc subproject, which does not look for MPI, and ends up in the branch of the if statement above where the paths hint is not used.
Still, even if this is not causing me a config error in scorpio, there really is no reason to ignore the user provided paths in the case where MPI is not found.

@bartgol 
Thanks for this patch. I can reproduce this issue on my laptop and your PR can fix it.

First, checkout and build mpi-serial from https://github.com/MCSclimate/mpi-serial

Next, try to configure scorpio with mpi-serial
```
git clone https://github.com/E3SM-Project/scorpio.git
cd scorpio
mkdir build
cd build
CC=gcc CXX=g++ FC=gfortran cmake -DPIO_USE_MPISERIAL=ON -DMPISERIAL_PATH=/home/wuda/3rdparty/mpi-serial -DNetCDF_C_PATHS=/home/wuda/3rdparty/netcdf -DNetCDF_Fortran_PATHS=/home/wuda/3rdparty/netcdf ..
```
The error message is:
```
-- Could NOT find NetCDF_C (missing: NetCDF_C_LIBRARY NetCDF_C_INCLUDE_DIR) 
CMake Error at src/clib/CMakeLists.txt:208 (message):
  Must have PnetCDF and/or NetCDF C libraries
```
With your PR applied, no CMake configure errors encountered.
```
-- Found NetCDF_C: /home/wuda/3rdparty/netcdf/lib/libnetcdf.so  
...
-- Found NetCDF_Fortran: /home/wuda/3rdparty/netcdf/lib/libnetcdff.so  
```

--- a/cmake/FindNetCDF.cmake
+++ b/cmake/FindNetCDF.cmake
@@ -47,11 +47,9 @@ foreach (NCDFcomp IN LISTS NetCDF_FIND_VALID_COMPONENTS)
             initialize_paths (NetCDF_${NCDFcomp}_PATHS
                               INCLUDE_DIRECTORIES ${MPI_${NCDFcomp}_INCLUDE_PATH}
                               LIBRARIES ${MPI_${NCDFcomp}_LIBRARIES})
-            find_package_component(NetCDF COMPONENT ${NCDFcomp}
-                                   PATHS ${NetCDF_${NCDFcomp}_PATHS})
-        else ()
-            find_package_component(NetCDF COMPONENT ${NCDFcomp})
         endif ()
+        find_package_component(NetCDF COMPONENT ${NCDFcomp}
+                               PATHS ${NetCDF_${NCDFcomp}_PATHS})
 
         # Continue only if component found
         if (NetCDF_${NCDFcomp}_FOUND)

